src/frontend/src/pages/PIAForm/index.tsx

```diff
@@ -396,11 +396,15 @@ const PIAFormPage = () => {
     if (isFirstSave) {
       setStalePia(updatedPia);
       setIsFirstSave(false);
-      navigate(
-        buildDynamicPath(routes.PIA_INTAKE_EDIT, {
-          id: updatedPia.id,
-        }),
-      );
+      if (pia?.id) {
+        navigate(pathname);
```

Comment:
why navigate if it is the same pathname? 🤔

Reply:
good catch, remove this check, just change path when the first time create pia